usr/share/rear/lib/uefi-functions.sh

```diff
@@ -37,20 +37,32 @@ function trim {
 }
 
 function build_bootx86_efi {
-    local gmkimage
-    if has_binary grub-mkimage; then
+    local gmkimage=""
+    if has_binary grub-mkimage ; then
         gmkimage=grub-mkimage
-    elif has_binary grub2-mkimage; then
+    elif has_binary grub2-mkimage ; then
+        # At least SUSE systems use 'grub2' prefixed names for GRUB2 programs:
         gmkimage=grub2-mkimage
     else
-        Log "Did not find grub-mkimage (cannot build bootx86.efi)"
-        return
+        # This build_bootx86_efi function is only called in output/ISO/Linux-i386/250_populate_efibootimg.sh
+        # which runs only if UEFI is used so that we simply error out here if we cannot make a bootable EFI image of GRUB2
+        # (normally a function should not exit out but return to its caller with a non-zero return code):
+        Error "Cannot make bootable EFI image of GRUB2 (neither grub-mkimage nor grub2-mkimage found)"
     fi
-    # as not all Linux distro's have the same grub modules present we verify what we have (see also https://github.com/rear/rear/pull/2001)
-    grub_modules=""
+    # grub-mkimage needs /usr/lib/grub/x86_64-efi/moddep.lst (cf. https://github.com/rear/rear/issues/1193)
+    # and at least on SUSE systems grub2-mkimage needs /usr/lib/grub2/x86_64-efi/moddep.lst (in 'grub2' directory)
+    # so that we error out if grub-mkimage or grub2-mkimage would fail when its moddep.lst is missing.
+    # Careful: usr/sbin/rear sets nullglob so that /usr/lib/grub*/x86_64-efi/moddep.lst gets empty if nothing matches
+    # and 'test -f' succeeds with empty argument so that we cannot use 'test -f /usr/lib/grub*/x86_64-efi/moddep.lst'
+    # also 'test -n' succeeds with empty argument but (fortunately/intentionally?) plain 'test' fails with empty argument:
+    test /usr/lib/grub*/x86_64-efi/moddep.lst || Error "$gmkimage would not make bootable EFI image of GRUB2 (no /usr/lib/grub*/x86_64-efi/moddep.lst file)"
+    # As not all Linux distros have the same GRUB2 modules present we verify what we have (see also https://github.com/rear/rear/pull/2001)
+    local grub_module=""
+    local grub_modules=""
     for grub_module in part_gpt part_msdos fat ext2 normal chain boot configfile linux linuxefi multiboot jfs iso9660 usb usbms usb_keyboard video udf ntfs all_video gzio efi_gop reboot search test echo btrfs ; do
         test "$( find /boot -type f -name "$grub_module.mod" 2>/dev/null )" && grub_modules="$grub_modules $grub_module"
     done
-    $gmkimage $v -O x86_64-efi -c $TMP_DIR/mnt/EFI/BOOT/embedded_grub.cfg -o $TMP_DIR/mnt/EFI/BOOT/BOOTX64.efi -p "/EFI/BOOT" $grub_modules
-    StopIfError "Error occurred during $gmkimage of BOOTX64.efi"
+    $gmkimage $v -O x86_64-efi -c $TMP_DIR/mnt/EFI/BOOT/embedded_grub.cfg -o $TMP_DIR/mnt/EFI/BOOT/BOOTX64.efi -p "/EFI/BOOT" $grub_modules && return 0
+    Error "Failed to make bootable EFI image of GRUB2 (error during $gmkimage of BOOTX64.efi)"
```

Comment:
First I was wondering why did you use Error() after `$gmkimage...`.  This would lead to complete stop of execution every time when build_bootx86_efi() is called, right ?
Then I've scrolled right and saw `&& return 0`.
I personally would prefer usage of StopIfError() at new line as scrolling all the way right to find this important piece of code could be confusing (at least for me :-)).

V.